theoplayer/how-to-guides/01-ads/13-custom-ssai-integration.md

```diff
@@ -0,0 +1,118 @@
+# Custom Server Side Ad Insertion
+
+THEOplayer provides a way to integrate with third-party advertisement providers,
+and have them report their ads and ad-related events through the THEOplayer APIs.
+
+## Android SDK
+
+### Prerequisites
+
+Your THEOplayer SDK needs to be on version 7.6.0 or higher.
+
+### Integration ID
+
+To make use of the custom server side ad insertion, first you need an integration ID which will identify the provided sources and ads to the specific ad provider.
+
+### CustomSsaiDescriptionSerializer
+
+Register a CustomSsaiDescriptionSerializer for the specific integration to be able to serialize/deserialize the
+CustomSsaiDescription that will be provided in the SourceDescription.
+
+```kotlin
+CustomSsaiDescriptionRegistry.register("integrationID", object : CustomSsaiDescriptionSerializer {
+
+    /**
+     * Reads a CustomSsaiDescription from the given JSON string.
+     *
+     * The customIntegration property in CustomSsaiDescription of the returned object
+     * **must** match the integration with which this serializer was registered.
+     */
+    override fun fromJson(json: String): CustomSsaiDescription {
```

Comment:
Most of the time, you'll also want to subclass `CustomSsaiDescription`. See [YospaceSsaiDescription](https://github.com/THEOplayer/android-connector/blob/8.8.3/connectors/yospace/src/main/java/com/theoplayer/android/connector/yospace/YospaceSsaiDescription.kt#L17-L57) for an example.